storm-core/src/jvm/org/apache/storm/scheduler/blacklist/BlacklistScheduler.java

```diff
@@ -0,0 +1,212 @@
+        LOG.info(conf.toString());
+        defaultScheduler=new DefaultScheduler();
+        defaultScheduler.prepare(conf);
+        _conf=conf;
+        if(_conf.containsKey(Config.BLACKLIST_SCHEDULER_TOLERANCE_TIME)){
+            toleranceTime=(Integer)_conf.get(Config.BLACKLIST_SCHEDULER_TOLERANCE_TIME);
+        }
+        if(_conf.containsKey(Config.BLACKLIST_SCHEDULER_TOLERANCE_COUNT)){
+            toleranceCount=(Integer)_conf.get(Config.BLACKLIST_SCHEDULER_TOLERANCE_COUNT);
+        }
+        if(_conf.containsKey(Config.BLACKLIST_SCHEDULER_RESUME_TIME)){
+            resumeTime=(Integer)_conf.get(Config.BLACKLIST_SCHEDULER_RESUME_TIME);
+        }
+        String reporterClassName=_conf.containsKey(Config.BLACKLIST_SCHEDULER_REPORTER)?(String)_conf.get(Config.BLACKLIST_SCHEDULER_REPORTER):"";
+        try {
+            reporter=(IReporter)Class.forName(reporterClassName).newInstance();
+        } catch (ClassNotFoundException e) {
+            LOG.error("Can't find blacklist reporter for name {}",reporterClassName);
+        } catch (InstantiationException e) {
+            LOG.error("Throw InstantiationException blacklist reporter for name {}", reporterClassName);
+        } catch (IllegalAccessException e) {
+            LOG.error("Throw illegalAccessException blacklist reporter for name {}", reporterClassName);
+        }
+
+        String strategyClassName=_conf.containsKey(Config.BLACKLIST_SCHEDULER_STRATEGY)?(String)_conf.get(Config.BLACKLIST_SCHEDULER_STRATEGY):"";
+        try {
+            blacklistStrategy=(IBlacklistStrategy)Class.forName(strategyClassName).newInstance();
+        } catch (ClassNotFoundException e) {
+            LOG.error("Can't find blacklist strategy for name {}",reporterClassName);
+        } catch (InstantiationException e) {
+            LOG.error("Throw InstantiationException blacklist strategy for name {}", reporterClassName);
+        } catch (IllegalAccessException e) {
+            LOG.error("Throw illegalAccessException blacklist strategy for name {}", reporterClassName);
+        }
+
+        nimbusMonitorFreqSecs=(Integer)_conf.get(Config.NIMBUS_MONITOR_FREQ_SECS);
+        blacklistStrategy.prepare(reporter,toleranceTime,toleranceCount,resumeTime,nimbusMonitorFreqSecs);
+
+        toleranceBuffer=new CircularBuffer<HashMap<String,Set<Integer>>>(toleranceTime/nimbusMonitorFreqSecs);
+        cachedSupervisors=new HashMap<>();
+
+
+    }
+
+    @Override
+    public void schedule(Topologies topologies, Cluster cluster) {
+        LOG.info("running Black List scheduler");
+        Map<String, SupervisorDetails> supervisors = cluster.getSupervisors();
+        for(Map.Entry<String,SupervisorDetails> entry:supervisors.entrySet()){
+            SupervisorDetails supervisorDetails=entry.getValue();
+            String supervisorName=entry.getKey();
+            Set<Integer> ports=supervisorDetails.getAllPorts();
+            LOG.info("supervisor: "+supervisorDetails.getHost()+" ports"+ports);
```

Comment:
SupervisorDetails is a class with all the supervisors info in it. It's not necessary to show all the info here. But if I override toString method, I have to put everything in it.